report-ng/app/src/components/classes/classes.ts

```diff
@@ -229,20 +318,110 @@ export class Classes extends AbstractViewModel {
+                const chip = {
+                    chipType: ChipType.CLEAR_ALL,
+                }
+                this._chipList.unshift(chip);
+            }
+            if(!this._chipList.filter(chip => chip.chipType === chipType).map(chip => chip.chipElement).includes(element)){
+                const chip: IChip = {
+                    chipType: chipType,
+                    chipElement: element
+                }
+                this._chipList.push(chip);
+            }
+        }
+
         this._filterOnce();
     }
+
+    private _removeFilter(chipType: ChipType, filterObject){
+        if(chipType == ChipType.CLEAR_ALL){
+            this._chipList = [];
+
+            delete this.queryParams.methods;
+            delete this.queryParams.q;
+            delete this.queryParams.status;
+            delete this.queryParams.class;
+            delete this.queryParams.config;
+            delete this.queryParams.failureAspect;
+            this.updateUrl(this.queryParams);
+
+            // reset class filter select options
+            this._filteredClassStatistics = [...this._executionStatistics.classStatistics.sort((a,b) => a.classIdentifier.localeCompare(b.classIdentifier))];
+
+        } else {
+
+            const index = this._chipList.map(chip => chip.chipElement).indexOf(filterObject)
+            this._chipList.splice(index, 1);
+
+            if(chipType == ChipType.CUSTOM_FILTER_TIMINGS || chipType == ChipType.CUSTOM_FILTER_FAILURE_ASPECTS){
+                // since the Custom Filter Chip only appears when navigating from another view to the tests view,
+                // we can use it for failureAspects and Methods,because they never appear together
+                delete this.queryParams.methods;
+                delete this.queryParams.failureAspect;
+            }
+
+            // insert classes that are not selected as filter anymore back into filter class select options
+            if(chipType == ChipType.CLASS && !this._filteredClassStatistics.some(stat => stat.classIdentifier == filterObject)){
+                this._filteredClassStatistics.push(this._executionStatistics.classStatistics.find(stat => stat.classIdentifier == filterObject));
+                this._filteredClassStatistics.sort((a,b) => a.classIdentifier.localeCompare(b.classIdentifier))
+            }
+
+            this.queryParams.q = this._convertQueryParams(this._chipList.filter(chip => chip.chipType === ChipType.CUSTOM_TEXT).map(chip => chip.chipElement), false);
+            this.updateUrl(this.queryParams);
+
+            if(this._chipList.length == 1){     // if there is only one chip left, it has to be removed because the last one will be the Clear All chip
+                this._chipList = [];
+            }
+        }
+        this._filter()
+    }
+
+    private _convertQueryParams(params, toArray: boolean){
```

Comment:
Could you explain what this method does exactly?
When you want to use multiple values for query parameters, the standard approach would be, to have multiple query parameters with the same name like:
`/my-url?q=Search1&q=Search2&class=Class1&class=Class3`
This should be supported by any modern framework. Please also see: https://stackoverflow.com/questions/724526/how-to-pass-multiple-parameters-in-a-querystring

Reply:
I used it to display multiple statuses like `status=1~2~3` instead of `status=%2C1%2C2` or similar, because the url encoding displayed arrays with filling characters like `%2C`. The `~` should be like a separator. I chose this with Martin to make the url shorter. It will not be necessary anymore when i implemented your suggestion

for the implementation i have a question and i could not really find an answer online: 
If i already have `status=4` in my url so i have `this.queryParams.status = 4`, how do i add a second one so that my url says `status=4&status=7 `?
I tried `this.queryParams.status = 7` but this just overwrites the `4` in the url, and `this.queryParams.status.push` or `append` does not work neither. Using this.queryParams["status"] also has the same results

Do you have any hints or am I misunderstanding something?